src/Modules/Meetings/Application/MeetingComments/AddMeetingCommentLike/AddMeetingCommentLikeCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using CompanyName.MyMeetings.BuildingBlocks.Application;
+using CompanyName.MyMeetings.Modules.Meetings.Application.Configuration.Commands;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.Comments;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingComments;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingGroups;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.MeetingMemberCommentLikes;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.Meetings;
+using CompanyName.MyMeetings.Modules.Meetings.Domain.Members;
+using MediatR;
+
+namespace CompanyName.MyMeetings.Modules.Meetings.Application.MeetingComments.AddMeetingCommentLike
+{
+    internal class AddMeetingCommentLikeCommandHandler : ICommandHandler<AddMeetingCommentLikeCommand>
+    {
+        private readonly IMeetingCommentRepository _meetingCommentRepository;
+        private readonly IMeetingGroupRepository _meetingGroupRepository;
+        private readonly IMeetingRepository _meetingRepository;
+        private readonly IMeetingMemberCommentLikesRepository _meetingMemberCommentLikesRepository;
+        private readonly IMemberContext _memberContext;
+        private readonly MeetingCommentService _meetingCommentService;
+
+        public AddMeetingCommentLikeCommandHandler(
+            IMeetingCommentRepository meetingCommentRepository,
+            IMeetingGroupRepository meetingGroupRepository,
+            IMeetingRepository meetingRepository,
+            IMeetingMemberCommentLikesRepository meetingMemberCommentLikesRepository,
+            IMemberContext memberContext,
+            MeetingCommentService meetingCommentService)
+        {
+            _meetingCommentRepository = meetingCommentRepository;
+            _meetingGroupRepository = meetingGroupRepository;
+            _meetingRepository = meetingRepository;
+            _memberContext = memberContext;
+            _meetingCommentService = meetingCommentService;
+            _meetingMemberCommentLikesRepository = meetingMemberCommentLikesRepository;
+        }
+
+        public async Task<Unit> Handle(AddMeetingCommentLikeCommand request, CancellationToken cancellationToken)
+        {
+            var meetingComment = await _meetingCommentRepository.GetByIdAsync(new MeetingCommentId(request.MeetingCommentId));
+            if (meetingComment == null)
+            {
+                throw new InvalidCommandException(new List<string> { "To add like the comment must exist." });
+            }
+
+            var meeting = await _meetingRepository.GetByIdAsync(meetingComment.GetMeetingId());
+
+            var meetingGroup = await _meetingGroupRepository.GetByIdAsync(meeting.GetMeetingGroupId());
+
+            var like = await _meetingCommentService.AddLikeAsync(meetingGroup, meetingComment, _memberContext.MemberId);
+
+            await _meetingMemberCommentLikesRepository.AddAsync(like);
```

Comment:
This is ok but I would like to point out that in real system it could be a performance problem. You are loading 2 another aggregates (meeting, meetingGroup) only to provide some information (not to protect invariants of these aggregates). Sometimes it is better to consider another approach - to load data via read model and pass to the method.

Reply:
I also think that load two other aggregates seems to much for such "light" action as liking but i did't find a better implementation. Thanks for explanation about Read model 👍 

>You are loading 2 another aggregates (meeting, meetingGroup) only to provide some information (not to protect invariants of these aggregates)

I need to protect "**Member** can like and unlike the comment" invariant. So MeetingGroup instance was needed to check if a liker is its member. I have looked at IsMemberOfGroup method of MeetingGroup aggregate again and understood that there isn't any logic there(just filter soft deleted members) so it's safe to check the invarinat other way.

Reply:
Is it ok to create `GetMeetingGroupMemberQuery` with `MemberId` and `MeetingId` values and pass result DTO to `MeetingComment` to check if group member with liker id exists or not?

Reply:
Yes it is ok to create separate data class, read it from read model and pass it to the aggregate method.

However, it is not DTO, it is just data, I like to name if with postfix Data and what is important **it is part of the Domain**. Domain can have data classes too - only we need to distinguish them from Entities and ValueObjects.

Reply:
Interesting insight about models with Data postfix, thanks! Now I've found few and their usages in the project.

>However, it is not DTO, it is just data

DTO's main use case is **transfer** so they should optimized to it(serialization): 
1. primitive types
2. no logic, no encapsulation - just container for data. 

Models with Data postfix are also containers for data but:
1. part of Domain model and so may contain domain objects(mostly ValueObjects, for example TypedIds)
2. aren't supposed to serialize
3. readonly

Am I right about these differences between the concepts?

Reply:
To continue discussion of concepts and names: what do you think about add ViewModel postfix for QueryHandlers result?

I think it will help to emphisize that the results are supposed to use in Presentation layer. On the contrary there are DTO which are part of Read model but are used in Application layer only.

Reply:
I would not call them ViewModel or View because you can have multiple consumers of your module API and API adapter is only one of them. Moreover, API can be consumed by non-GUI process like CLI.

What we return is a just data. I call it with postfix Dto to separate them from other objects. They are transferred between the layers and returned to API consumer. However, they can be called with postfix Data too in my opinion.

Reply:
Ok